search-commons/src/main/java/no/unit/nva/search2/resource/ResourceStreamBuilders.java

```diff
@@ -96,22 +94,14 @@ public Stream<Map.Entry<ResourceParameter, QueryBuilder>> fundingQuery(ResourceP
         return Functions.queryToEntry(key, query);
     }
 
-
     public Stream<Map.Entry<ResourceParameter, QueryBuilder>> subUnitIncludedQuery(ResourceParameter key) {
-        var query =
-            parameters.get(EXCLUDE_SUBUNITS).asBoolean()
-                ? termQuery(extractTermPath(key), parameters.get(key).as())
-                : termQuery(extractMatchPath(key), parameters.get(key).as());
-
-        return Functions.queryToEntry(key, query);
-    }
-
-    private String extractTermPath(ResourceParameter key) {
-        return key.searchFields(KEYWORD_TRUE).findFirst().orElseThrow();
-    }
+        var searchKey = parameters.get(ResourceParameter.EXCLUDE_SUBUNITS).asBoolean()
```

Comment:
is this in correct order?

Reply:
Well, yes, but I understand why you are asking. Let me see if I can improve readability.